cmd/messages_text.go

```diff
@@ -80,21 +80,32 @@ func (mt *MessagesText) reset() {
 	mt.Highlight()
 }
 
+// Region tags are square brackets that contain a region ID in double quotes
+// https://pkg.go.dev/github.com/rivo/tview#hdr-Regions_and_Highlights
+func (mt *MessagesText) startRegion(m discord.Message) {
+	fmt.Fprintf(mt, `["%s"]`, m.ID)
+}
+
+// Tags with no region ID ([""]) don't start new regions. They can therefore be used to mark the end of a region.
+func (mt *MessagesText) endRegion() {
+	fmt.Fprint(mt, `[""]`)
+}
+
 func (mt *MessagesText) createMessage(m discord.Message) {
+	mt.startRegion(m)
 	if cfg.HideBlockedUsers {
 		isBlocked := discordState.UserIsBlocked(m.Author.ID)
 		if isBlocked {
 			fmt.Fprintln(mt, "[:red:b]Blocked message[:-:-]")
+			mt.endRegion()
```

Comment:
Defer the `endRegion` function instead of calling it every time you return from the function:
```go
mt.startRegion(m.ID)
defer mt.endRegion()
```

Reply:
deferred that, and Fprintln as well (blocked msg uses Fprint now)